### PR TITLE
Feedback on porting guide

### PR DESCRIPTION
**Pull request by bridadan**
I've included quite a bit of comments and corrections here. **This PR isn't intended to be merged as is,** its more like a starting point of feedback that can be referenced.

I only covered the "Porting process" section in the porting guide. The guide is quite extensive and has lots of potential!

### CODE REVIEW

**Comment by iriark01** [comment]
Adding Don, as it'll be easier if everything's in the same PR rather than clashing ones.

**Comment by AnotherButler** [comment]
@bridadan @linlingao Do y'all have time this afternoon to sit with me to resolve the comments from @donatieng ?

File: docs/porting/porting_full_process/built_in_tests.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ To run the Mbed OS built-in tests, you need to have ported and verified at least
 
 - DAPLink.
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/porting/porting_full_process/built_in_tests.md]: DAPLink or compatible interface firmware.

File: docs/porting/porting_full_process/built_in_tests.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ To run the Mbed OS built-in tests, you need to have ported and verified at least
...
 
-    <span class="notes">If DAPLink is still under development, please [use manual tests](#manual-testing).</span>
+    <span class="notes">If DAPLink is still under development, please [use manual tests](#manual-testing)<!-- This link doesn't work -->.</span>
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/porting/porting_full_process/built_in_tests.md]: For testing, you need mass storage device and serial.

File: docs/porting/porting_full_process/built_in_tests.md [hunk 3]
```diff
@@ -158,8 +161,11 @@ You may want to run manual tests, for example if DAPLink is still under developm
     ```
     mbed export -i gcc_arm -m <new_target>
     ```
-
+<!-- gcc_arm is not a valid exporter. A list of valid exporters are here: https://github.com/ARMmbed/mbed-os/blob/master/tools/export/__init__.py#L36-L60 -->
+<!-- If this is actually meant to use Eclipse, a valid eclipse exporter needs to be used (and ensure that the exporter is working before committing it to the docs) -->
 1. Open the project with pyOCD (using the same configuration you used [when you initially set up pyOCD]( Creating GDB pyOCD debug configuration).
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/porting/porting_full_process/built_in_tests.md]: Configure your IDE and/or debugger to program your device [only for manual tests/examples]
Note: Update throughout doc to show pyOCD is not required

File: docs/porting/porting_full_process/built_in_tests.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ To run the Mbed OS built-in tests, you need to have ported and verified at least
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/porting/porting_full_process/built_in_tests.md]: 👍 Done

File: docs/porting/porting_full_process/built_in_tests.md [hunk 5]
```diff
@@ -48,10 +48,11 @@ The board under test needs to be supported in mbedls for automated tests to work
 If the official mbedls pip package hasn't been released yet, you need to direct pip to use your local directory (which includes the  code changes to support the new board):
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/porting/porting_full_process/built_in_tests.md]: Done 👍

File: docs/porting/porting_full_process/built_in_tests.md [hunk 6]
```diff
@@ -61,17 +62,20 @@ If the official mbedls pip package hasn't been released yet, you need to direct
     }
     ```
 
+    Where `"33000000e062afa300000000000000000000000097969902"` is the correct target id.
+
 ### Compiling and running tests
 
 1. Compile your tests:
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/porting/porting_full_process/built_in_tests.md]: Done 👍

File: docs/porting/porting_full_process/built_in_tests.md [hunk 7]
```diff
@@ -184,23 +190,17 @@ To build and run the Mbed OS tests:
 
 1. Build the tests:
 
+   <!-- any reason why we're doing a clean build here everytime? I'd really recommend dropping this one, build times will be very high -->
    ```
-   mbed test --compile -m <new_target> -t gcc_arm -c
+   mbed test -m <new_target> -t gcc_arm --compile -c
    ```
 
    You'll see some build errors. These errors should reduce and eventually disappear as more HAL components are ported.
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/porting/porting_full_process/built_in_tests.md]: Done 👍

**Comment by AnotherButler** [comment]
@donatieng I've added clarification that this document shows how to port a new MCU target with pyOCD and DAPLink. Porting instructions for existing targets and other methods can come in another document after 5.11.

**Comment by iriark01** [comment]
@donatieng and @bridadan I think it's best to close this PR and start a fresh review - everything here is out of date. 
Any chance we can finish this today?

**Comment by iriark01** [comment]
Yes please

**Comment by iriark01** [comment]
Yes please

**Comment by donatieng** [comment]
Cool, give me a ping when it's ready for re-review!

**Comment by bridadan** [comment]
Ok rebased!

**Comment by AnotherButler** [comment]
ping @donatieng Ready for rereview

**Comment by iriark01** [comment]
@donatieng and @AnotherButler I'm merging this so I can make the 5.11 branch for testing; please finish your edits on 5.11 this week and then, at your leisure, copy them to development.

**Pull request merged**